// vitepress-plugin-rss/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/index.ts
var src_exports = {};
__export(src_exports, {
  RssPlugin: () => RssPlugin,
  genFeed: () => genFeed,
  svgIcon: () => svgIcon
});
module.exports = __toCommonJS(src_exports);
var import_theme_shared2 = require("@sugarat/theme-shared");

// src/node.ts
var import_node_fs2 = __toESM(require("fs"));
var import_node_path2 = __toESM(require("path"));
var import_node_crypto2 = __toESM(require("crypto"));
var import_feed = require("feed");
var import_theme_shared = require("@sugarat/theme-shared");
var import_markdown_it_container = __toESM(require("markdown-it-container"));

// src/transform.ts
var import_node_fs = __toESM(require("fs"));
var import_node_path = __toESM(require("path"));
var import_node_crypto = __toESM(require("crypto"));
var import_sharp = __toESM(require("sharp"));
function removeZeroWidthSpace(html) {
  return html.replaceAll("&ZeroWidthSpace;", "");
}
function svgToBase64(html) {
  return html.replaceAll(/<svg.*?<\/svg>/g, (svg) => {
    const base64 = Buffer.from(svg).toString("base64");
    return `<img src="data:image/svg+xml;base64,${base64}" alt="svg image" />`;
  });
}
function svgToUrl(html, outputDir, assetsBasePath) {
  return __async(this, null, function* () {
    const regex = /<svg[\s\S]*?<\/svg>/g;
    const matches = Array.from(html.matchAll(regex));
    if (matches.length === 0) {
      return html;
    }
    const results = yield Promise.all(matches.map((match) => __async(this, null, function* () {
      const svg = match[0];
      const fileName = yield svg2png(svg, outputDir);
      return {
        match,
        fileName
      };
    })));
    let lastIndex = 0;
    let resultHtml = "";
    for (const { match, fileName } of results) {
      const start = match.index;
      const end = start + match[0].length;
      resultHtml += html.slice(lastIndex, start);
      if (fileName) {
        const url = `${assetsBasePath}/${import_node_path.default.basename(outputDir)}/${fileName}`;
        resultHtml += `<img src="${url}" alt="svg image" />`;
      } else {
        resultHtml += match[0];
      }
      lastIndex = end;
    }
    resultHtml += html.slice(lastIndex);
    return resultHtml;
  });
}
function svg2png(svg, outDir) {
  return __async(this, null, function* () {
    if (!import_node_fs.default.existsSync(outDir)) {
      yield import_node_fs.default.promises.mkdir(outDir, { recursive: true });
    }
    const hash = import_node_crypto.default.createHash("md5").update(svg).digest("hex");
    const fileName = `${hash}.png`;
    const filePath = import_node_path.default.join(outDir, fileName);
    if (import_node_fs.default.existsSync(filePath)) {
      return fileName;
    }
    try {
      yield (0, import_sharp.default)(Buffer.from(svg)).png().toFile(filePath);
      return fileName;
    } catch (error) {
      console.error("Failed to convert SVG to PNG:", error);
      return null;
    }
  });
}
function injectStyle(html, style) {
  return `<style>${style}</style>${html}`;
}

// src/node.ts
var imageRegex = /!\[.*?\]\((.*?)\s*(".*?")?\)/;
var htmlCache = /* @__PURE__ */ new Map();
function emptyArrayToUndefined(arr) {
  return (arr == null ? void 0 : arr.length) ? arr : void 0;
}
function normalizeEnclosure(frontmatter) {
  const enclosureObj = frontmatter == null ? void 0 : frontmatter.enclosure;
  if (typeof enclosureObj === "string") {
    return {
      url: enclosureObj
    };
  }
  if (enclosureObj == null ? void 0 : enclosureObj.url) {
    return {
      url: enclosureObj.url,
      length: (enclosureObj == null ? void 0 : enclosureObj.length) || 0,
      type: enclosureObj == null ? void 0 : enclosureObj.type,
      title: enclosureObj == null ? void 0 : enclosureObj.title,
      duration: enclosureObj == null ? void 0 : enclosureObj.duration
    };
  }
  if (frontmatter == null ? void 0 : frontmatter.enclosure_url) {
    return {
      url: frontmatter.enclosure_url,
      length: (frontmatter == null ? void 0 : frontmatter.enclosure_length) || 0,
      type: frontmatter == null ? void 0 : frontmatter.enclosure_type,
      title: frontmatter == null ? void 0 : frontmatter.enclosure_title,
      duration: frontmatter == null ? void 0 : frontmatter.enclosure_duration
    };
  }
  return void 0;
}
function stringifyForHash(val) {
  return JSON.stringify(val, (k, v) => {
    if (typeof v === "function")
      return v.toString();
    if (v instanceof RegExp)
      return v.toString();
    return v;
  });
}
function getCacheKey(config, rssOptions, filepath, content, url) {
  const basename = url ? import_node_path2.default.basename(url) : import_node_path2.default.basename(filepath);
  const markdownStr = stringifyForHash((config == null ? void 0 : config.markdown) || {});
  const rssOpsStr = stringifyForHash(rssOptions || {});
  const payload = `${basename}|${markdownStr}|${rssOpsStr}`;
  const hash = import_node_crypto2.default.createHash("md5").update(`${content}|${payload}`).digest("hex");
  return `${basename}_${hash}.html`;
}
function getCacheDir(config) {
  return __async(this, null, function* () {
    const cacheDir = import_node_path2.default.join(config.cacheDir || import_node_path2.default.join(config.root, ".vitepress/cache"), "rss");
    yield import_node_fs2.default.promises.mkdir(cacheDir, { recursive: true });
    return cacheDir;
  });
}
function getCachedHtml(config, rssOptions, filepath, content, url) {
  return __async(this, null, function* () {
    try {
      const cacheDir = yield getCacheDir(config);
      const cacheKey = getCacheKey(config, rssOptions, filepath, content, url);
      const cachePath = import_node_path2.default.join(cacheDir, cacheKey);
      if (import_node_fs2.default.existsSync(cachePath)) {
        const cachedHtml = yield import_node_fs2.default.promises.readFile(cachePath, "utf-8");
        return cachedHtml;
      }
    } catch (error) {
    }
    return null;
  });
}
function setCachedHtml(config, rssOptions, filepath, content, html, url) {
  return __async(this, null, function* () {
    try {
      const cacheDir = yield getCacheDir(config);
      const cacheKey = getCacheKey(config, rssOptions, filepath, content, url);
      const cachePath = import_node_path2.default.join(cacheDir, cacheKey);
      yield import_node_fs2.default.promises.writeFile(cachePath, html, "utf-8");
    } catch (error) {
    }
  });
}
function getMdRender(config, rssOptions) {
  return __async(this, null, function* () {
    const { createMarkdownRenderer, disposeMdItInstance } = yield import("vitepress");
    const mdOptions = __spreadValues(__spreadProps(__spreadValues({}, config.markdown), {
      // 默认行号（RSS 样式会渲染异常）
      lineNumbers: false
    }), rssOptions.markdownOptions);
    disposeMdItInstance();
    const md = yield createMarkdownRenderer(
      config.srcDir,
      mdOptions,
      config.site.base,
      config.logger
    );
    md.use((md2) => {
      const fence = md2.renderer.rules.fence;
      md2.renderer.rules.fence = (...args) => {
        var _a;
        const [tokens, idx] = args;
        const token = tokens[idx];
        const title = (_a = token.info.match(/\[(.*)\]/)) == null ? void 0 : _a[1];
        return title == null ? fence(...args) : `<tr><td>${title}</td><td>${fence(...args)}</td></tr>`;
      };
    });
    md.use(import_markdown_it_container.default, "code-group", {
      render(tokens, idx) {
        return tokens[idx].nesting === 1 ? "<table><tbody>" : "</tbody></table>\n";
      }
    });
    return md;
  });
}
var cachePageData;
function getPostsData(_0, _1) {
  return __async(this, arguments, function* (config, rssOptions, assetsMap = []) {
    var _a, _b, _c, _d;
    const endParseConfig = (0, import_theme_shared.debugTime)("endParseConfig");
    const { ignoreHome = true, ignorePublish = false, renderHTML = true, assetsBaseUrl = rssOptions.baseUrl } = rssOptions;
    if (!cachePageData) {
      cachePageData = (0, import_theme_shared.getVitePressPages)(config);
    }
    endParseConfig();
    let posts = [];
    const fileContentPromises = cachePageData.reduce((prev, f) => {
      const { isDynamic, dynamicRoute, filepath } = f;
      const contentPromise = isDynamic && dynamicRoute ? Promise.resolve((0, import_theme_shared.renderDynamicMarkdown)(filepath, dynamicRoute == null ? void 0 : dynamicRoute.params, dynamicRoute == null ? void 0 : dynamicRoute.content)) : import_node_fs2.default.promises.readFile(filepath, "utf-8");
      prev[f.page] = {
        contentPromise,
        datePromise: (0, import_theme_shared.getFileLastModifyTime)(f.filepath, config.cacheDir)
      };
      return prev;
    }, {});
    const endParsePageData = (0, import_theme_shared.debugTime)("endParsePageData");
    for (const page of cachePageData) {
      const { contentPromise, datePromise } = fileContentPromises[page.page];
      let fileContent = yield contentPromise;
      const { data: frontmatter, excerpt, content } = (0, import_theme_shared.grayMatter)(fileContent, {
        excerpt: true
      });
      if (!frontmatter.title) {
        frontmatter.title = (0, import_theme_shared.getDefaultTitle)(content);
      }
      frontmatter.title = `${frontmatter.title}`;
      const date = yield frontmatter.date || datePromise;
      frontmatter.date = (0, import_theme_shared.formatDate)(date);
      frontmatter.published = frontmatter.published && (0, import_theme_shared.formatDate)(frontmatter.published);
      frontmatter.description = (_b = yield (_a = rssOptions == null ? void 0 : rssOptions.renderExpect) == null ? void 0 : _a.call(rssOptions, content, __spreadValues({}, frontmatter))) != null ? _b : frontmatter.description || excerpt || (0, import_theme_shared.getTextSummary)(content, 100);
      frontmatter.cover = ((_d = frontmatter.cover) != null ? _d : (_c = fileContent.match(imageRegex)) == null ? void 0 : _c[1]) || "";
      const relativeImages = getRelativeImagesFromFileContent(content, page.filepath.replace(config.srcDir, ""));
      if (relativeImages.length > 0) {
        relativeImages.forEach((v) => {
          const [originValue, relativePath] = v;
          const absolutePath = (0, import_theme_shared.joinPath)(config.srcDir, `/${relativePath}`);
          const asset = assetsMap.find((item) => {
            const assetPath = (0, import_theme_shared.joinPath)(config.srcDir, `/${item.originalFileName}`);
            return assetPath === absolutePath;
          });
          if (asset) {
            const imageUrl = (0, import_theme_shared.normalizeUrl)(`${assetsBaseUrl}${(0, import_theme_shared.joinPath)("/", asset.fileName)}`);
            v.push(imageUrl);
            fileContent = fileContent.replaceAll(originValue, imageUrl);
          }
        });
      }
      if (frontmatter.cover) {
        const exist = relativeImages.find((v) => v[0] === frontmatter.cover);
        if (exist) {
          frontmatter.cover = exist[2];
        }
      }
      const targetPage = (0, import_theme_shared.normalizePath)(
        page.rewritePath ? page.rewritePath : page.page
      );
      const url = (0, import_theme_shared.joinPath)(config.site.base, targetPage).replace(/(^|\/)index\.md$/, "$1").replace(/\.md$/, config.cleanUrls ? "" : ".html");
      posts.push({
        filepath: page.filepath,
        env: page.env,
        fileContent,
        description: frontmatter.description,
        date: frontmatter.date,
        title: frontmatter.title,
        url,
        frontmatter
      });
    }
    endParsePageData();
    posts = posts.filter((p) => {
      if (p.frontmatter.layout === "home" && ignoreHome) {
        return false;
      }
      if (p.frontmatter.publish === false && !ignorePublish)
        return false;
      return true;
    });
    if (rssOptions == null ? void 0 : rssOptions.filter) {
      posts = posts.filter(rssOptions.filter);
    }
    posts.sort((a, b) => {
      const dateA = a.frontmatter.published || a.date;
      const dateB = b.frontmatter.published || b.date;
      return +new Date(dateB) - +new Date(dateA);
    });
    if (void 0 !== (rssOptions == null ? void 0 : rssOptions.limit) && (rssOptions == null ? void 0 : rssOptions.limit) > 0) {
      posts.splice(rssOptions.limit);
    }
    const endRenderHTML = (0, import_theme_shared.debugTime)("endRenderHTML");
    const md = yield getMdRender(config, rssOptions);
    yield Promise.all(posts.map((post) => __async(this, null, function* () {
      var _a2, _b2;
      const { fileContent, filepath, env, url } = post;
      if (!htmlCache.has(url)) {
        let html = "";
        const cachedHtml = rssOptions.cache !== false ? yield getCachedHtml(config, rssOptions, filepath, fileContent, url) : null;
        if (cachedHtml) {
          html = cachedHtml;
        } else {
          if (renderHTML === true) {
            html = md.render(fileContent, env);
          } else if (typeof renderHTML === "function") {
            html = yield renderHTML(fileContent, config, rssOptions);
          }
          html = removeZeroWidthSpace(html);
          if ((_a2 = rssOptions == null ? void 0 : rssOptions.markdownOptions) == null ? void 0 : _a2.style) {
            html = injectStyle(html, rssOptions.markdownOptions.style);
          }
          if (rssOptions == null ? void 0 : rssOptions.transform) {
            const transforms = [rssOptions == null ? void 0 : rssOptions.transform].flat();
            for (const transform of transforms) {
              html = yield transform(html, config);
            }
          }
          if (html) {
            yield setCachedHtml(config, rssOptions, filepath, fileContent, html, url);
          }
        }
        if ((_b2 = rssOptions == null ? void 0 : rssOptions.markdownOptions) == null ? void 0 : _b2.svg2img) {
          let target = rssOptions.markdownOptions.svg2img;
          target = target === true ? "base64" : target;
          if (target === "base64") {
            html = svgToBase64(html);
          } else if (target === "png") {
            const assetsBaseUrl2 = (rssOptions == null ? void 0 : rssOptions.assetsBaseUrl) || rssOptions.baseUrl;
            const outDir = (0, import_theme_shared.joinPath)(config.outDir, `/${config.assetsDir}`);
            html = yield svgToUrl(html, outDir, assetsBaseUrl2);
          }
        }
        if (html) {
          htmlCache.set(url, html);
        }
      }
    })));
    endRenderHTML();
    return posts;
  });
}
function genFeed(config, rssOptions, assetsMap, localePrefix) {
  return __async(this, null, function* () {
    var _b, _c, _d;
    if (!rssOptions)
      return;
    const { baseUrl, filename } = rssOptions;
    let urlPrefix = "";
    if (localePrefix && localePrefix !== "root") {
      urlPrefix = `${localePrefix}/`;
    }
    const homepageUrl = (0, import_theme_shared.normalizeUrl)(`${baseUrl}${config.site.base}${urlPrefix}`);
    const _a = rssOptions, { renderHTML } = _a, restOps = __objRest(_a, ["renderHTML"]);
    const feed = new import_feed.Feed(__spreadValues({
      id: homepageUrl,
      link: homepageUrl
    }, restOps));
    const endGetPostsData = (0, import_theme_shared.debugTime)("genFeed:getPostsData");
    const posts = yield getPostsData(config, rssOptions, assetsMap);
    endGetPostsData();
    for (const post of posts) {
      const { title, description, date, frontmatter, url } = post;
      const author = emptyArrayToUndefined(
        [(_b = frontmatter.author) != null ? _b : rssOptions.author].flat().filter(Boolean).map((a) => {
          var _a2, _b2;
          if (typeof a === "string") {
            return ((_a2 = rssOptions.authors) == null ? void 0 : _a2.find((v) => v.name === a)) || { name: a };
          }
          if ((a == null ? void 0 : a.name) && !a.email && !a.link && !a.avatar) {
            const found = (_b2 = rssOptions.authors) == null ? void 0 : _b2.find((v) => v.name === a.name);
            return found ? __spreadValues(__spreadValues({}, a), found) : a;
          }
          return a;
        })
      );
      const category = emptyArrayToUndefined(frontmatter.category);
      const link = `${baseUrl}${url}`;
      const guid = (frontmatter == null ? void 0 : frontmatter.guid) || link;
      const published = (frontmatter == null ? void 0 : frontmatter.published) && new Date(frontmatter == null ? void 0 : frontmatter.published);
      const enclosure = normalizeEnclosure(frontmatter);
      feed.addItem({
        title,
        guid,
        link,
        description,
        content: (_c = htmlCache.get(post.url)) != null ? _c : "",
        author,
        category,
        published,
        enclosure,
        image: (frontmatter == null ? void 0 : frontmatter.cover) ? new URL(frontmatter == null ? void 0 : frontmatter.cover, baseUrl).href : "",
        date: new Date(date)
      });
    }
    const RSSFilename = filename || "feed.rss";
    if (!import_node_fs2.default.existsSync(config.outDir)) {
      import_node_fs2.default.mkdirSync(config.outDir, { recursive: true });
    }
    const RSSFilepath = import_node_path2.default.join(config.outDir, RSSFilename);
    yield import_node_fs2.default.promises.writeFile(RSSFilepath, feed.rss2(), "utf-8");
    if ((_d = rssOptions.log) != null ? _d : true) {
      console.log("\u{1F389} RSS generated", RSSFilename);
      console.log("rss filepath:", RSSFilepath);
      console.log("rss url:", (0, import_theme_shared.normalizeUrl)(`${baseUrl}${config.site.base + RSSFilename}`));
      console.log("included", posts.length, "posts");
    }
  });
}
function isBase64ImageURL(url) {
  const regex = /^data:image\/[a-z]+;base64,/;
  return regex.test(url);
}
function getRelativeImagesFromFileContent(content, route) {
  const images = [...content.matchAll(/!\[.*?\]\((.*?)\s*(".*?")?\)/g)].map((v) => v[1]) || [];
  return images.map((originValue) => {
    const isHTTPSource = originValue && originValue.startsWith("http");
    if (isHTTPSource || isBase64ImageURL(originValue)) {
      return [originValue, ""];
    }
    const paths = (0, import_theme_shared.joinPath)("/", route).split("/");
    paths.splice(paths.length - 1, 1);
    const relativePath = (0, import_theme_shared.normalizePath)(
      originValue.startsWith("/") ? originValue : import_node_path2.default.join(paths.join("/") || "", originValue)
    );
    return [originValue, (0, import_theme_shared.joinPath)("/", relativePath)];
  }).filter((v) => {
    return v[1].startsWith("/");
  });
}

// src/index.ts
var svgIcon = '<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 448 512"><title>RSS</title><path d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zM112 416c-26.51 0-48-21.49-48-48s21.49-48 48-48s48 21.49 48 48s-21.49 48-48 48zm157.533 0h-34.335c-6.011 0-11.051-4.636-11.442-10.634c-5.214-80.05-69.243-143.92-149.123-149.123c-5.997-.39-10.633-5.431-10.633-11.441v-34.335c0-6.535 5.468-11.777 11.994-11.425c110.546 5.974 198.997 94.536 204.964 204.964c.352 6.526-4.89 11.994-11.425 11.994zm103.027 0h-34.334c-6.161 0-11.175-4.882-11.427-11.038c-5.598-136.535-115.204-246.161-251.76-251.76C68.882 152.949 64 147.935 64 141.774V107.44c0-6.454 5.338-11.664 11.787-11.432c167.83 6.025 302.21 141.191 308.205 308.205c.232 6.449-4.978 11.787-11.432 11.787z" fill="currentColor"></path></svg>';
function SocialIcon(rssOptions, base = "/") {
  const icon = typeof rssOptions.icon === "string" ? rssOptions.icon : svgIcon;
  return {
    icon: {
      svg: icon
    },
    link: rssOptions.url || (0, import_theme_shared2.normalizeUrl)(`${rssOptions.baseUrl}${base}${rssOptions.filename || "feed.rss"}`),
    ariaLabel: (rssOptions == null ? void 0 : rssOptions.ariaLabel) || "RSS"
  };
}
function inheritLocaleConfig(target, global) {
  const excludeKeys = ["locales"];
  Object.keys(global).forEach((key) => {
    const k = key;
    if (excludeKeys.includes(k)) {
      return;
    }
    if (typeof target[k] === "undefined") {
      target[k] = global[k];
    }
  });
}
function RssPlugin(rssOptions) {
  let resolveConfig;
  let vitepressConfig;
  let assetsDir;
  let assetsMap = [];
  const pluginOps = {
    name: "vitepress-plugin-rss",
    enforce: "pre",
    configResolved(config) {
      return __async(this, null, function* () {
        var _a2, _b, _c, _d;
        if (resolveConfig) {
          return;
        }
        resolveConfig = config;
        vitepressConfig = config.vitepress;
        assetsDir = vitepressConfig.assetsDir;
        const VPConfig = config.vitepress;
        if (!VPConfig) {
          return;
        }
        yield (0, import_theme_shared2.cacheAllGitTimestamps)(VPConfig.srcDir);
        const localesConfig = [];
        if ((_a2 = rssOptions.icon) != null ? _a2 : true) {
          VPConfig.site.themeConfig.socialLinks = [
            SocialIcon(rssOptions, VPConfig.site.base),
            ...((_b = VPConfig.site.themeConfig) == null ? void 0 : _b.socialLinks) || []
          ];
          if ((rssOptions == null ? void 0 : rssOptions.locales) && ((_c = VPConfig.site) == null ? void 0 : _c.locales)) {
            Object.keys((_d = VPConfig.site) == null ? void 0 : _d.locales).forEach((locale) => {
              var _a3, _b2, _c2, _d2;
              const rssCfg = (_a3 = rssOptions == null ? void 0 : rssOptions.locales) == null ? void 0 : _a3[locale];
              if (rssCfg && ((_b2 = rssCfg == null ? void 0 : rssCfg.icon) != null ? _b2 : true)) {
                inheritLocaleConfig(rssCfg, rssOptions);
                const _tcfg = (_c2 = VPConfig.site.locales[locale]) == null ? void 0 : _c2.themeConfig;
                if (!_tcfg) {
                  VPConfig.site.locales[locale].themeConfig = {};
                }
                VPConfig.site.locales[locale].themeConfig.socialLinks = [
                  SocialIcon(rssCfg, VPConfig.site.base),
                  ...(_tcfg == null ? void 0 : _tcfg.socialLinks) || []
                ];
                if (!rssCfg.filter) {
                  const otherLocales = Object.keys(((_d2 = VPConfig.site) == null ? void 0 : _d2.locales) || {}).filter((l) => l !== "root");
                  const base = VPConfig.site.base || "/";
                  if (locale === "root") {
                    rssCfg.filter = (value) => {
                      return !otherLocales.some((l) => value.url.startsWith(`${base}${l}/`));
                    };
                  } else {
                    rssCfg.filter = (value) => {
                      return value.url.startsWith(`${base}${locale}/`);
                    };
                  }
                }
                localesConfig.push([locale, rssCfg]);
              }
            });
          }
        }
        const selfBuildEnd = VPConfig.buildEnd;
        VPConfig.buildEnd = (siteConfig) => __async(this, null, function* () {
          var _a3;
          yield selfBuildEnd == null ? void 0 : selfBuildEnd(siteConfig);
          const okMark = "\x1B[32m\u2713\x1B[0m";
          console.time(`${okMark} generating RSS`);
          for (const [localePrefix, _rssOptions] of localesConfig) {
            yield genFeed(siteConfig, _rssOptions, assetsMap, localePrefix);
          }
          yield genFeed(siteConfig, rssOptions, assetsMap);
          console.timeEnd(`${okMark} generating RSS`);
          if ((_a3 = rssOptions.log) != null ? _a3 : true) {
            console.log();
          }
        });
      });
    },
    generateBundle(_, bundle) {
      assetsMap = Object.entries(bundle).filter(([key]) => {
        return key.startsWith(assetsDir);
      }).map(([_2, value]) => {
        return value;
      }).filter((v) => v.type === "asset");
    }
  };
  return pluginOps;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RssPlugin,
  genFeed,
  svgIcon
});
